src/Sylius/Component/Core/Provider/LastYearSalesSummaryProvider.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Provider;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+
+final class LastYearSalesSummaryProvider implements LastYearSalesSummaryProviderInterface
+{
+    /** @var ServiceLocator */
+    private $locator;
+
+    public function __construct(ServiceLocator $locator)
+    {
+        $locator = $this->locator;
+    }
+
+
+
+    public function getSalesSummary(ChannelInterface $channel, \DateTimeInterface $startDate, \DateTimeInterface $endDate): array
+    {
+        return [];
```

Comment:
```suggestion
        if (!$this->locator->has($connection) {
            $provider = $this->locator->get('default');
        } else {
            $provider = $this->locator->get($connection);
        }
        
        return $provider-> getSalesSummary($channel, $start, $end);
```

Where `$connection` is taken from the doctrine entity manager. (and without else)

Reply:
The locator logic could be extracted to separate service to make it easier to test (we should pass connection as an argument there)